### ISSUE TITLE
[QUERY] Confirm behavior of Azure-AsyncOperation for PATCH/PUT with final-state-via set

### ISSUE THREAD

**Issue by jhendrixMSFT** [Azure/azure-sdk-for-python#36936]
Pollers in some languages assume that for `Azure-AsyncOperation` LROs initiated with a `PATCH`/`PUT` that the "final get" is made against the originating URL and not the value specified in the `final-state-via` setting.

Can you please confirm if Python pollers have the same behavior?

This potentially impacts the [azure-mgmt-quota](https://github.com/Azure/azure-sdk-for-python/tree/main/sdk/quota/azure-mgmt-quota) SDK as the latest preview version contains an `Azure-AsyncOperation` LRO initiated with a `PUT` that also sets the `final-state-via` to `Location`.

**Comment by lmazuel** [comment]
Python does not honor final-state-via for anything else than POST. Final-State-Via was designed initially as a way to disambiguate when a service team was returning both AzureAsyncOperation and Location at the same time for the initial response on a POST call. It was never intended to be used outside of this context 🙁

We did an experiment a few years ago to add PUT in Python azure-core, and everything just broke suddenly, we had to revert that azure-core release. We realized that service teams don't know how to use this option, and a huge number of them in Swagger are just plain wrong. We just don't know, because we don't read them...
